_posts/2021-08-15-interviews.markdown

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+title: "Code Interviews"
+date: 2021-08-10 20:18:32 -0500
+categories: interviews
+---
+
+For most of us, interviewing is hard. And annoying, frustrating, and stressful. Most interviews these days consist of a string of 1-hour interviews covering:
+
+- Programming (sometimes 2 of these).
+- Systems design.
+- 1-2 specialization-specific interviews: ML, Data Engineering, Frontend, Distributed Systems, etc.
+- An interview with a hiring manager. Sometimes this is about "leadership" or other general topics depending on the role.
+
+The programming interview is probably the most hated. I personally don't like being on either side of these interviews. Leetcode-style problems that test _CS fundamentals_ can be fun puzzles for some people, and are arguably a decent evaluation of junior engineers without industry experience, but it's a common complaint among senior engineers that they have to study material that they haven't used in years to get a job where they won't use it. This tweet, from the author of [Homebrew](https://brew.sh/), lives in infamy.
```

Comment:
> a common complaint among senior engineers that they have to study material that they haven't used in years to get a job where they won't use it

I wonder if you have any personal anecdotes that could be interesting to mention here? like a time when you were asked in an interview to prove you knew algorithm X in a role that would never use it